sdk/identity/Azure.Identity/src/EnvironmentCredential.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    public class EnvironmentCredential : TokenCredential
+    {
+        private TokenCredential _credential = null;
+
+        public EnvironmentCredential()
+            : this(null)
+        {
+        }
+
+        public EnvironmentCredential(IdentityClientOptions options)
+        {
+            string tenantId = Environment.GetEnvironmentVariable("AZURE_TENANT_ID");
```

Comment:
Might be a good place to use `IConfiguration` rather than specifically going for ENV vars that way the consumer can specify the binding and mapping of configuration data.

Reply:
We're still working on the configuration story for Azure.Core and client libraries.  This direct read of the environment variables is just a temporary place holder for the prototype